algopy_testing/src/algopy/op.py

```diff
@@ -168,3 +178,398 @@ def vrf_verify(
+
+
+def bitlen(a: Bytes | UInt64 | bytes | int, /) -> UInt64:
+    int_value = int.from_bytes(as_bytes(a)) if (isinstance(a, Bytes | bytes)) else as_int64(a)
+    return UInt64(int_value.bit_length())
+
+
+def bsqrt(a: BigUInt | int, /) -> BigUInt:
+    a = as_int512(a)
+    return BigUInt(math.isqrt(a))
+
+
+def btoi(a: Bytes | bytes, /) -> UInt64:
+    a_bytes = as_bytes(a)
+    if len(a_bytes) > 8:
+        raise ValueError(f"btoi arg too long, got [{len(a_bytes)}]bytes")
+    return UInt64(int.from_bytes(a_bytes))
+
+
+def bzero(a: UInt64 | int, /) -> Bytes:
+    a = as_int64(a)
+    if a > MAX_BYTES_SIZE:
+        raise ValueError("bzero attempted to create a too large string")
+    return Bytes(b"\x00" * a)
+
+
+def divmodw(
+    a: UInt64 | int, b: UInt64 | int, c: UInt64 | int, d: UInt64 | int, /
+) -> tuple[UInt64, UInt64, UInt64, UInt64]:
+    i = _uint128_to_int(a, b)
+    j = _uint128_to_int(c, d)
+    d = i // j
+    m = i % j
+    return _int_to_uint128(d) + _int_to_uint128(m)
+
+
+def divw(a: UInt64 | int, b: UInt64 | int, c: UInt64 | int, /) -> UInt64:
+    i = _uint128_to_int(a, b)
+    c = as_int64(c)
+    return UInt64(i // c)
+
+
+def err() -> None:
+    raise ArithmeticError("err opcode executed")
+
+
+def exp(a: UInt64 | int, b: UInt64 | int, /) -> UInt64:
+    a = as_int64(a)
+    b = as_int64(b)
+    if a == b and a == 0:
+        raise ArithmeticError("0^0 is undefined")
+    return UInt64(a**b)
+
+
+def expw(a: UInt64 | int, b: UInt64 | int, /) -> tuple[UInt64, UInt64]:
+    a = as_int64(a)
+    b = as_int64(b)
+    if a == b and a == 0:
+        raise ArithmeticError("0^0 is undefined")
+    result = a**b
+    return _int_to_uint128(result)
+
+
+def extract(a: Bytes | bytes, b: UInt64 | int, c: UInt64 | int, /) -> Bytes:
+    a = a if (isinstance(a, Bytes)) else Bytes(a)
+    b = as_int64(b)
+    c = as_int64(c)
+    if b > len(a):
+        raise ArithmeticError(f"extraction start {b} is beyond length")
+    if b + c > len(a):
+        raise ArithmeticError(f"extraction end {b + c} is beyond length")
+    return a[slice(b, b + c)]
+
+
+def extract_uint16(a: Bytes | bytes, b: UInt64 | int, /) -> UInt64:
+    result = extract(a, b, 2)
+    result_int = int.from_bytes(result.value)
+    return UInt64(result_int)
+
+
+def extract_uint32(a: Bytes | bytes, b: UInt64 | int, /) -> UInt64:
+    result = extract(a, b, 4)
+    result_int = int.from_bytes(result.value)
+    return UInt64(result_int)
+
+
+def extract_uint64(a: Bytes | bytes, b: UInt64 | int, /) -> UInt64:
+    result = extract(a, b, 8)
+    result_int = int.from_bytes(result.value)
+    return UInt64(result_int)
+
+
+def getbit(a: Bytes | UInt64 | bytes | int, b: UInt64 | int, /) -> UInt64:
+    if isinstance(a, Bytes | bytes):
+        return _getbit_bytes(a, b)
+    if isinstance(a, UInt64 | int):
+        a_bytes = _uint64_to_bytes(a)
+        return _getbit_bytes(a_bytes, b, "little")
+    raise ValueError("Unknown type for argument a")
```

Comment:
Use TypeError here

Reply:
It is fixed in the commit be7f752396a4563859209c063f7a03577a4f680f.